src/github_changelog/git.clj

```diff
@@ -8,28 +9,37 @@
            (org.eclipse.jgit.lib Repository Ref)
            (org.eclipse.jgit.revwalk RevCommit)))
 
-(def git-path name-from-uri)
-
-(defn clone-or-load [uri]
-  (let [path (git-path uri)]
-    (try
-      (git/load-repo path)
-      (catch FileNotFoundException _
-        (git/git-clone uri path)))))
+(defn gen-url [{:keys [git user repo]}]
+  (format "%s/%s/%s.git" (strip-trailing git) user repo))
 
-(defn clone [uri]
-  (let [repo (clone-or-load uri)]
-    (git/git-fetch-all repo)
-    repo))
+(def git-path name-from-uri)
 
-(defn get-merge-sha [^Repository repo ^Ref tag]
+(defn- clone-or-load [uri dir]
+  (try
+    (git/load-repo dir)
+    (catch FileNotFoundException _
+      (git/git-clone uri dir))))
+
+(defn- refresh [^Repository repo]
+  (git/git-fetch-all repo)
+  repo)
+
+(defn clone [config]
+  (let [{:keys [git-url dir update?]
+         :or   {git-url (gen-url config)
+                dir     (git-path git-url)
+                update? true}} config
+        repo                   (clone-or-load git-url dir)]
```

Comment:
Because that expands to:
```clojure
(if update? refresh identity (clone-or-load git-url dir))
```
You can do what you wanted with the `as->` macro, but I don't think you get too much out of it:
```clojure
(macroexpand '(as-> (clone-or-load git-url dir) repo (if update? (refresh repo) repo)))
;; (let* [repo (clone-or-load git-url dir) repo (if update? (refresh repo) repo)] repo)
```